lib/scottkit/play.rb

```diff
@@ -6,27 +6,32 @@ def play
       prepare_to_play
 
       while true
+        # TODO: Find out if this can move to the bottom of the loop. If so we
+        # could probably turn the outter loop into `while !finished?`
         run_matching_actions(0, 0)
-        if @need_to_look
-          actually_look
-        end
+
+        actually_look if @need_to_look
 
         return @finished if finished?
 
         print "Tell me what to do ? "
         if !(line = gets)
+          # TODO: Find out when we would we end up here?
```

Comment:
Any thoughts on this bit right here? I couldn't figure out how to trigger it.

Reply:
`gets` returns `nil` at end of file, so this line handles when a user quits the game by typing CTRL-D (or whatever their terminal software uses to signal EOF), and of course the equally important case when you're piping a sequence of commands into `scottkit`.

So I've replaced your TODO with a comment to that effect.